lib/puma/cluster.rb

```diff
@@ -456,32 +456,34 @@ def run
 
             if read.wait_readable([0, @next_check - Time.now].max)
               req = read.read_nonblock(1)
+              next unless req
 
-              @next_check = Time.now if req == "!"
-              next if !req || req == "!"
+              if req == Puma::Const::PipeRequest::WAKEUP
+                next @next_check = Time.now
```

Comment:
WDYT?

```suggestion
                @next_check = Time.now
                next
```

It would be more like how it was before this change (Yeah, I know, it looks like nothing it using the result from the iterations of this `while` loop, so `next @next_check = Time.now` doesn't hurt but maybe it makes someone think more about it than they need to, like I did now 😄 )

Reply:
On second thought I agree, actioned in [these changes](https://github.com/puma/puma/compare/513e57fac98b85a0b7a2d3a1cb4ea5393665039b..3d82649aa6cf47dc87086c292dafd589e2d9e959).